core-web/libs/portlets/edit-ema/portlet/src/lib/store/dot-uve.store.spec.ts

```diff
@@ -0,0 +1,1646 @@
+                    } as unknown as ActivatedRouteSnapshot
+                } as unknown as ActivatedRoute);
+
+                store.load(VTL_BASE_QUERY_PARAMS);
+
+                expect(router.navigate).not.toHaveBeenCalled();
+            });
+
+            it('should not navigate to content when the layout is disable by page.canEdit and current route is not layout', () => {
+                jest.spyOn(dotPageApiService, 'get').mockImplementation(() =>
+                    of({
+                        ...MOCK_RESPONSE_VTL,
+                        page: {
+                            ...MOCK_RESPONSE_VTL.page,
+                            canEdit: false
+                        }
+                    })
+                );
+
+                jest.spyOn(activatedRoute, 'firstChild', 'get').mockReturnValue({
+                    snapshot: {
+                        url: [
+                            {
+                                path: 'rules',
+                                parameters: {},
+                                parameterMap: {} as unknown as ParamMap
+                            }
+                        ]
+                    } as unknown as ActivatedRouteSnapshot
+                } as unknown as ActivatedRoute);
+
+                store.load(VTL_BASE_QUERY_PARAMS);
+
+                expect(router.navigate).not.toHaveBeenCalled();
+            });
+
+            it('should reload the store with the same queryParams', () => {
+                const getPageSpy = jest.spyOn(dotPageApiService, 'get');
+
+                store.reload();
+
+                expect(getPageSpy).toHaveBeenCalledWith(store.params());
+            });
+        });
+    });
+
+    describe('withLayout', () => {
+        describe('withComputed', () => {
+            describe('$layoutProps', () => {
+                it('should return the layout props', () => {
+                    expect(store.$layoutProps()).toEqual({
+                        containersMap: mapContainerStructureToDotContainerMap(
+                            MOCK_RESPONSE_HEADLESS.containers
+                        ),
+                        layout: MOCK_RESPONSE_HEADLESS.layout,
+                        template: {
+                            identifier: MOCK_RESPONSE_HEADLESS.template.identifier,
+                            themeId: MOCK_RESPONSE_HEADLESS.template.theme
+                        },
+                        pageId: MOCK_RESPONSE_HEADLESS.page.identifier
+                    });
+                });
+            });
+        });
+
+        describe('withMethods', () => {
+            it('should update the layout', () => {
+                const layout = {
+                    ...MOCK_RESPONSE_HEADLESS.layout,
+                    title: 'New layout'
+                };
+
+                store.updateLayout(layout);
+
+                expect(store.pageAPIResponse().layout).toEqual(layout);
+            });
+        });
+    });
+
+    describe('withEditor', () => {
```

Comment:
I mean, we could call the `UVEStore` in different files to test each `signalStoreFeature` or create a `SignalStore` Mock with the necessary properties to test each feature

Reply:
I might not be in this PR due to time, but it's worth testing because we have a mega test file with over 1000 lines. What are your thoughts @KevinDavilaDotCMS @fmontes ?